test/i18n_viz_test.rb

```diff
@@ -51,3 +51,36 @@ class I18nVizJavascriptIntegrationTest < ActionDispatch::IntegrationTest
     assert page.has_css?("#i18n_viz_tooltip")
   end
 end
+
+class I18nVizDynamicUrlTest < ActionDispatch::IntegrationTest
+  setup do
+    Capybara.current_driver = Capybara.javascript_driver # :selenium by default
+
+    # monkey patch external_tool_url
+    I18nViz::Middleware.class_eval do
+      alias orig_external_tool_url external_tool_url
+      def external_tool_url
+        lambda { |x| "http://example.com/?s=" }
+      end
+    end
+  end
+
+  teardown do
+    # revert monkey patch
+    I18nViz::Middleware.class_eval do
+      alias external_tool_url orig_external_tool_url
+    end
+  end
+
+  test 'translate with dynamic external url parameter' do
+    visit "/test?i18n_viz=1"
+
+    assert !page.has_content?("--hello--") # should be removed by js
```

Comment:
I would remove all the assertions that are not part of checking the dynamic tool url.  since the other things are already tested elsewhere.